File: src/chain-observers/classifiers/dynamic-global-properties-classifier.ts
import { CollectorClassifierBase } from "./collector-classifier-base";

export interface IDynamicGlobalPropertiesData {
  headBlockNumber: number;
  currentWitness: string;
  headBlockTime: Date;
  headBlockId: string;
  downvotePoolPercent: number;
}

export class DynamicGlobalPropertiesClassifier extends CollectorClassifierBase<{}, IDynamicGlobalPropertiesData> {}

File: python/workerbee/chain_observers/classifiers/dynamic_global_properties_classifier.py
from __future__ import annotations

from ..classifier_results import DynamicGlobalPropertiesData
from .collector_classifier_base import CollectorClassifierBase


class DynamicGlobalPropertiesClassifier(CollectorClassifierBase[DynamicGlobalPropertiesData]):
    """IDynamicGlobalPropertiesData:
    head_block_number: int
    current_witness: str
    head_block_time: datetime
    head_block_id: str
    downvote_pool_percent: float
    """

    pass
